app/src/main/java/com/example/memeitupapp/ui/gridmemes/GirdMemesViewModel.kt

```diff
@@ -15,11 +16,10 @@ import kotlinx.coroutines.launch
 import kotlinx.coroutines.withContext
 
 class GridMemesViewModel : ViewModel() {
-
-    private val memesRepository: MemeRepository = MemeRepository()
+    @VisibleForTesting
+    var memesRepository: MemeRepository = MemeRepository()
```

Comment:
repository should be injected by parameter and be private.
val viewmodel = viewModel(MemeRepo())

also it should be private.

you may use contracts

Reply:
1-. no, the repository must be in the Model. 

You should create the ViewModel in the fragment/activity like this
val viewmodel = viewModel(Model(MemeRepo()))

Reply:
fixed the issues, added contracts and Model